p2p/p2p.go

```diff
@@ -163,16 +166,24 @@ func (s *Service) SubscribePeerConnectednessChanged(ctx context.Context) (<-chan
 	return ch, nil
 }
 
+// Run starts the p2p service. Calling any other function before run is undefined behaviour
 func (s *Service) Run(ctx context.Context) error {
-	var err error
+	err := func() error {
+		defer s.runLock.Unlock()
```

Comment:
I assumed it will be called only once. In our flow, there is no reason for it to be called multiple times. If you think we shouldnt assume that, I can try to come up with a solution.

Reply:
Ok, agree if there are problems in the future, we will rewrite it.